src/server_manager/web_app/ui_components/outline-server-list.ts

```diff
@@ -0,0 +1,62 @@
+// Copyright 2021 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {customElement, html, LitElement, property} from 'lit-element';
+import {repeat} from 'lit-html/directives/repeat.js';
+import {ServerView} from './outline-server-view.js';
+
+export interface ServerListEntry {
+  id: string;
+  name: string;
+  isManaged: boolean;
+  isSynced: boolean;
+}
+
+@customElement('outline-server-list')
+export class OutlineServerList extends LitElement {
+  @property({type: Array}) serverList: ServerListEntry[];
+  @property({type: String}) selectedServerId: string;
+  @property({type: Function}) localize: Function;
+  @property({type: String}) language: string;
+
+  render() {
+    if (!this.serverList) {
+      return;
+    }
+    return html`<div>${repeat(this.serverList, e => e.id, e => html`
```

Comment:
Let's add typing for the list elements?

Reply:
serverList is already typed, so are the parameters of repeat.